### DISCUSSION TITLE
Neovim: disable specific diagnostics (e.g. types)

### DISCUSSION THREAD

**Discussion by rob32** [phpactor/phpactor#2670] [Q&A] [answered]
Is there a way to disable specific diagnostics like types (missing return type etc. or function/var not found)?

I've spent hours trying out and researching otherwise I wouldn't be asking.

Many thanks in advance.

---

settings:

```lua
["phpactor"] = function()
        lspconfig["phpactor"].setup {
          root_dir = function(_)
            return vim.loop.cwd()
          end,
          init_options = {
            ["language_server_phpstan.enabled"] = false,
            ["language_server_psalm.enabled"] = false,
            -- ["completion_worse.completor.type.enabled"] = false, //  does not help 
          },
        }
      end,
```

**Comment by mamazu** [comment, marked as answer]
There is the `language_server.diagnostic_providers` (it's an array) where you can list all providers that should be enabled. If you don't know what values are possible in the list just write a string that's not valid and it'll list all providers that are possible. Hope it helps.

**Reply by rob32** [reply]
thank you very much, that helped!
